from-env-derive/src/field.rs

```diff
@@ -0,0 +1,221 @@
+
+    _attrs: Vec<syn::Attribute>,
+
+    span: proc_macro2::Span,
+}
+
+impl TryFrom<&syn::Field> for Field {
+    type Error = syn::Error;
+
+    fn try_from(field: &syn::Field) -> Result<Self, syn::Error> {
+        let mut optional = false;
+        let mut env_var = None;
+        let mut infallible = false;
+        let mut desc = None;
+
+        field
+            .attrs
+            .iter()
+            .filter(|attr| attr.path().is_ident("from_env"))
+            .for_each(|attr| {
+                let _ = attr.parse_nested_meta(|meta| {
+                    if meta.path.is_ident("optional") {
+                        optional = true;
+                        return Ok(());
+                    }
+                    if meta.path.is_ident("var") {
+                        env_var = Some(meta.value()?.parse::<LitStr>()?);
+                        return Ok(());
+                    }
+                    if meta.path.is_ident("desc") {
+                        desc = Some(meta.value()?.parse::<LitStr>()?.value());
+                        return Ok(());
+                    }
+                    if meta.path.is_ident("infallible") {
+                        infallible = true;
+                    }
+                    Ok(())
+                });
+            });
+
+        if desc.is_none() && env_var.is_some() {
+            return Err(syn::Error::new(
+                field.span(),
+                "Missing description for field. Use `#[from_env(desc = \"DESC\")]`",
+            ));
+        }
+
+        let field_type = field.ty.clone();
+        let field_name = field.ident.clone();
+        let span = field.span();
+
+        Ok(Field {
+            env_var,
+            field_name,
+            field_type,
+            optional,
+            infallible,
+            desc,
+            _attrs: field
+                .attrs
+                .iter()
+                .filter(|attr| !attr.path().is_ident("from_env"))
+                .cloned()
+                .collect(),
+            span,
+        })
+    }
+}
+
+impl Field {
+    pub(crate) fn trait_name(&self) -> TokenStream {
+        self.env_var
+            .as_ref()
+            .map(|_| quote! { FromEnvVar })
+            .unwrap_or(quote! { FromEnv })
+    }
+
+    pub(crate) fn as_trait(&self) -> TokenStream {
+        let field_trait = self.trait_name();
+        let field_type = &self.field_type;
+
+        quote! { <#field_type as #field_trait> }
+    }
+
+    pub(crate) fn assoc_err(&self) -> TokenStream {
+        let as_trait = self.as_trait();
+
+        quote! { #as_trait::Error }
+    }
+
+    pub(crate) fn field_name(&self, idx: usize) -> Ident {
+        if let Some(field_name) = self.field_name.as_ref() {
+            return field_name.clone();
+        }
+
+        let n = format!("field_{}", idx);
+        syn::parse_str::<Ident>(&n)
+            .map_err(|_| syn::Error::new(self.span, "Failed to create field name"))
+            .unwrap()
+    }
```

Comment:
Right, so for tuple structs, we'd get instead of the struct prop name, `field_0, field_1...,field_n` and so forth? This seems fine to me to support tuple structs although I'm not sure in which cases you'd want to do a config with a tuple struct.